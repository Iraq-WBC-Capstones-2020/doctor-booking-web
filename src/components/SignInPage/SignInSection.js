import React, { useState, useContext, useEffect } from 'react';
import { Row, Col, Form, Button } from 'react-bootstrap';
import { firebaseFunctions } from '../../firebaseFunctions';
import { DoctorContext, ACTIONS } from '../../DoctorContext';
function SignInSection() {
  const [, dispatch] = useContext(DoctorContext);
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const handleEmail = (e) => {
    setEmail(e.target.value);
  };
  const handlePassword = (e) => {
    setPassword(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    firebaseFunctions.signIn(email, password).then(() => {
      dispatch({ type: ACTIONS.IS_SIGNED_IN, isSignedIn: true });
    });
  };

  return (
    <div className="d-flex align-items-center">
      <div className="welcome-rectangle d-none d-md-flex align-items-center justify-content-center">
        <h1>
          WELCOME <br /> BACK
        </h1>
      </div>
      <div className="signIn-rectangle">
        <Row>
          <Col className="justify-content-center d-flex text-white" sm="12">
            <h3>SIGN IN</h3>
          </Col>
          <Col
            className="justify-content-center d-flex mt-5 text-white"
            sm="12"
          >
            <Form
              className="justify-content-center d-flex flex-column"
              onSubmit={handleSubmit}
            >
              <Form.Group onChange={handleEmail} controlId="email">
                <Form.Control size="lg" type="email" placeholder="E-mail" />
              </Form.Group>
              <Form.Group
                onChange={handlePassword}
                controlId="password"
                className="mt-2"
              >
                <Form.Control
                  size="lg"
                  type="password"
                  placeholder="Password"
                />
              </Form.Group>
              <Button
                className="signin-btn btn-lg mt-3"
                variant="primary"
                type="submit"
              >
                Sign In
              </Button>
              <p className="mt-3">
                Don{"'"}t have an account? <a href="#!">Sign Up</a>
              </p>
            </Form>
          </Col>
        </Row>
      </div>
    </div>
  );
}

export default SignInSection;
