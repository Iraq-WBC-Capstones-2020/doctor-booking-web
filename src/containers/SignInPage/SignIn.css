.page-container {
  height: 80vh;
}

.welcome-rectangle {
  height: 350px;
  width: 275px;
  background: #014d87;
  -webkit-box-shadow: 0px 11px 14px rgba(1, 77, 135, 0.3);
  box-shadow: 0px 11px 14px rgba(1, 77, 135, 0.3);
  border-radius: 10px 0 0 10px;
}

.welcome-rectangle h1 {
  color: white;
}

.signIn-rectangle {
  max-width: 350px;
  min-height: 480px;
  background: #026fc2;
  -webkit-box-shadow: 0px 15px 17px rgba(2, 111, 194, 0.3);
  box-shadow: 0px 15px 17px rgba(2, 111, 194, 0.3);
  border-radius: 10px;
}

.signIn-rectangle input {
  font-size: 14px;
  width: 250px;
  height: 45px;
}

.signIn-rectangle p {
  color: white;
  margin: 0 auto;
}

.signIn-rectangle a {
  color: white;
}

.signIn-rectangle a:hover {
  color: white;
}

.signIn-rectangle h3 {
  margin-top: 4rem;
}

.signin-btn {
  background-color: white;
  color: #026fc2;
  width: 100px;
  height: 40px;
  font-size: 14px;
  margin: 0 auto;
}

.signin-btn:hover {
  background-color: #014d87;
  color: white;
}
/*# sourceMappingURL=SignIn.css.map */
