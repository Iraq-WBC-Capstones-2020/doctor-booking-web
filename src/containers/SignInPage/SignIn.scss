@import '../../styles/variables';
.welcome-rectangle {
  height: 350px;
  width: 275px;
  background: #014d87;
  box-shadow: 0px 11px 14px rgba(1, 77, 135, 0.3);
  border-radius: 10px 0 0 10px;
  h1 {
    color: white;
  }
}
.signin-rectangle {
  max-width: 350px;
  min-height: 480px;
  background: #026fc2;
  box-shadow: 0px 15px 17px rgba(2, 111, 194, 0.3);
  border-radius: 10px;
  input {
    font-size: 14px;
    width: 250px;
    height: 45px;
  }
  p {
    color: white;
    margin: 0 auto;
  }
  a {
    color: white;
    &:hover {
      color: white;
    }
  }
  h3 {
    margin-top: 4rem;
  }
}
.signin-btn {
  background-color: white;
  color: $primaryColor;
  &:hover {
    background-color: $darkerBlue;
    color: white;
  }
  width: 100px;
  height: 40px;
  font-size: 14px;
  margin: 0 auto;
}
