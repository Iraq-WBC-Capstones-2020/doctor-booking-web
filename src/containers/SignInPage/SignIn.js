import React from 'react';
import { Container } from 'react-bootstrap';
import SignInSection from '../../components/SignInPage/SignInSection';
function SignIn() {
  return (
    <Container className="d-flex justify-content-center mt-5">
      <SignInSection />
    </Container>
  );
}

export default SignIn;
