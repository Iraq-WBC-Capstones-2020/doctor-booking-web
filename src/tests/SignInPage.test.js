import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SignInSection from '../components/SignInPage/SignInSection';
import renderer from 'react-test-renderer';
import SignIn from '../containers/SignInPage/SignIn';
test('sign in section is rendering ', () => {
  render(<SignInSection />);
});

test('sign in page renders correctly', () => {
  const tree = renderer.create(<SignIn />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('e-mail should be string', () => {
  const signIn = render(<SignInSection />);
  console.log(signIn);
  const input = signIn.getByPlaceholderText('E-mail');
  fireEvent.change(input, { target: { value: 'anything@anything.com' } });
  expect(input.value).toBe('anything@anything.com');
});
