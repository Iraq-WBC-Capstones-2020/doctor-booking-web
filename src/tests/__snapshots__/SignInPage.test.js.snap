// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign in page renders correctly 1`] = `
<div
  className="d-flex align-items-center"
>
  <div
    className="welcome-rectangle d-none d-md-flex align-items-center justify-content-center"
  >
    <h1>
      WELCOME 
      <br />
       BACK
    </h1>
  </div>
  <div
    className="signIn-rectangle"
  >
    <div
      className="row"
    >
      <div
        className="justify-content-center d-flex text-white col-sm-12"
      >
        <h3>
          SIGN IN
        </h3>
      </div>
      <div
        className="justify-content-center d-flex mt-5 text-white col-sm-12"
      >
        <form
          className="justify-content-center d-flex flex-column"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
            onChange={[Function]}
          >
            <input
              className="form-control form-control-lg"
              id="email"
              placeholder="E-mail"
              type="email"
            />
          </div>
          <div
            className="mt-2 form-group"
            onChange={[Function]}
          >
            <input
              className="form-control form-control-lg"
              id="password"
              placeholder="Password"
              type="password"
            />
          </div>
          <button
            className="signin-btn btn-lg mt-3 btn btn-primary"
            disabled={false}
            type="submit"
          >
            Sign In
          </button>
          <p
            className="mt-3"
          >
            Don
            '
            t have an account? 
            <a
              href="/register"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </p>
        </form>
      </div>
    </div>
  </div>
</div>
`;
